[Bug]: Milvus crashed when inserting empty string for varchar field

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:2.1.1
- Deployment mode(standalone or cluster):
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus 2.1.1
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```


### Current Behavior

Milvus crashed when inserting empty string for varchar field.

The crash log:
```
[2022/08/29 02:44:56.850 +00:00] [INFO] [datacoord/services.go:1019] ["[flush state] unflushed segment ids"] [segmentIDs="[435615456365903873,435615456365903874]"] [len=2]
[2022/08/29 02:44:56.850 +00:00] [INFO] [proxy/impl.go:3585] ["received get flush state response"] [response="status:<> "]
[2022/08/29 02:44:57.353 +00:00] [INFO] [proxy/impl.go:3571] ["received get flush state request"] [request="segmentIDs:435615456365903873 segmentIDs:435615456365903874 "]
[2022/08/29 02:44:57.353 +00:00] [INFO] [datacoord/services.go:996] ["DataCoord receive get flush state request"] [segmentIDs="[435615456365903873,435615456365903874]"] [len=2]
[2022/08/29 02:44:57.353 +00:00] [INFO] [datacoord/services.go:1019] ["[flush state] unflushed segment ids"] [segmentIDs="[435615456365903873,435615456365903874]"] [len=2]
[2022/08/29 02:44:57.354 +00:00] [INFO] [proxy/impl.go:3585] ["received get flush state response"] [response="status:<> "]
[2022/08/29 02:44:57.856 +00:00] [INFO] [proxy/impl.go:3571] ["received get flush state request"] [request="segmentIDs:435615456365903873 segmentIDs:435615456365903874 "]
[2022/08/29 02:44:57.857 +00:00] [INFO] [datacoord/services.go:996] ["DataCoord receive get flush state request"] [segmentIDs="[435615456365903873,435615456365903874]"] [len=2]
[2022/08/29 02:44:57.857 +00:00] [INFO] [datacoord/services.go:1019] ["[flush state] unflushed segment ids"] [segmentIDs="[435615456365903873,435615456365903874]"] [len=2]
[2022/08/29 02:44:57.857 +00:00] [INFO] [proxy/impl.go:3585] ["received get flush state response"] [response="status:<> "]
[2022/08/29 02:44:58.360 +00:00] [INFO] [proxy/impl.go:3571] ["received get flush state request"] [request="segmentIDs:435615456365903873 segmentIDs:435615456365903874 "]
[2022/08/29 02:44:58.360 +00:00] [INFO] [datacoord/services.go:996] ["DataCoord receive get flush state request"] [segmentIDs="[435615456365903873,435615456365903874]"] [len=2]
[2022/08/29 02:44:58.361 +00:00] [INFO] [datacoord/services.go:1019] ["[flush state] unflushed segment ids"] [segmentIDs="[435615456365903873,435615456365903874]"] [len=2]
[2022/08/29 02:44:58.361 +00:00] [INFO] [proxy/impl.go:3585] ["received get flush state response"] [response="status:<> "]
[2022/08/29 02:44:58.863 +00:00] [INFO] [proxy/impl.go:3571] ["received get flush state request"] [request="segmentIDs:435615456365903873 segmentIDs:435615456365903874 "]
[2022/08/29 02:44:58.864 +00:00] [INFO] [datacoord/services.go:996] ["DataCoord receive get flush state request"] [segmentIDs="[435615456365903873,435615456365903874]"] [len=2]
[2022/08/29 02:44:58.864 +00:00] [INFO] [datacoord/services.go:1019] ["[flush state] unflushed segment ids"] [segmentIDs="[435615456365903873,435615456365903874]"] [len=2]
[2022/08/29 02:44:58.865 +00:00] [INFO] [proxy/impl.go:3585] ["received get flush state response"] [response="status:<> "]
[2022/08/29 02:44:58.960 +00:00] [ERROR] [datanode/flow_graph_insert_buffer_node.go:402] ["insertBufferNode flushBufferData failed, err = All attempts results:\nattempt #1:can't add empty string into payload\nattempt #2:can't add empty string into payload\nattempt #3:can't add empty string into payload\nattempt #4:can't add empty string into payload\nattempt #5:can't add empty string into payload\n"] [stack="github.com/milvus-io/milvus/internal/datanode.(*insertBufferNode).Operate\n\t/go/src/github.com/milvus-io/milvus/internal/datanode/flow_graph_insert_buffer_node.go:402\ngithub.com/milvus-io/milvus/internal/util/flowgraph.(*nodeCtx).work\n\t/go/src/github.com/milvus-io/milvus/internal/util/flowgraph/node.go:102"]
[2022/08/29 02:44:58.960 +00:00] [ERROR] [datanode/flow_graph_insert_buffer_node.go:402] ["insertBufferNode flushBufferData failed, err = All attempts results:\nattempt #1:can't add empty string into payload\nattempt #2:can't add empty string into payload\nattempt #3:can't add empty string into payload\nattempt #4:can't add empty string into payload\nattempt #5:can't add empty string into payload\n"] [stack="github.com/milvus-io/milvus/internal/datanode.(*insertBufferNode).Operate\n\t/go/src/github.com/milvus-io/milvus/internal/datanode/flow_graph_insert_buffer_node.go:402\ngithub.com/milvus-io/milvus/internal/util/flowgraph.(*nodeCtx).work\n\t/go/src/github.com/milvus-io/milvus/internal/util/flowgraph/node.go:102"]
panic: insertBufferNode flushBufferData failed, err = All attempts results:
attempt #1:can't add empty string into payload
attempt #2:can't add empty string into payload
attempt #3:can't add empty string into payload
attempt #4:can't add empty string into payload
attempt #5:can't add empty string into payload


goroutine 2329 [running]:
github.com/milvus-io/milvus/internal/datanode.(*insertBufferNode).Operate(0xc00247c4b0, 0xc003648a50, 0x1, 0x1, 0x434ba80, 0xc00365e580, 0x0)
	/go/src/github.com/milvus-io/milvus/internal/datanode/flow_graph_insert_buffer_node.go:403 +0x40f5
github.com/milvus-io/milvus/internal/util/flowgraph.(*nodeCtx).work(0xc000e4a380)
	/go/src/github.com/milvus-io/milvus/internal/util/flowgraph/node.go:102 +0x23b
created by github.com/milvus-io/milvus/internal/util/flowgraph.(*nodeCtx).Start
	/go/src/github.com/milvus-io/milvus/internal/util/flowgraph/node.go:70 +0x70

```

### Expected Behavior
Milvus can accept empty string.


_No response_

### Steps To Reproduce


1. download v2.1.1
wget https://github.com/milvus-io/milvus/releases/download/v2.1.1/milvus-standalone-docker-compose.yml -O docker-compose.yml

2. start a standalone server
docker-compose up -d

3. use this script to test
```markdown
import random
from pymilvus import (
    connections,
    FieldSchema, CollectionSchema, DataType,
    Collection,
    utility
)

if __name__ == '__main__':
    connections.connect(host=_HOST, port=_PORT)

    dim = 128
    field1 = FieldSchema(name="id", dtype=DataType.INT64, description="int64", is_primary=True)
    field2 = FieldSchema(name="vec", dtype=DataType.FLOAT_VECTOR, description="float vector", dim=dim)
    field3 = FieldSchema(name="str", dtype=DataType.VARCHAR, description="string",
                         max_length=128, is_primary=False)
    schema = CollectionSchema(fields=[field1, field2, field3], description="collection description")
    collection = Collection(name="TEST", data=None, schema=schema)

    num = 1000
    data = [
        [i for i in range(num)],  # id field
        [[random.random() for _ in range(dim)] for _ in range(num)],  # vector field
        ["" for _ in range(num)],  # string field
    ]
    print("Start insert ...")
    collection.insert(data)
    print("Insert finished")

    print("Row count:", collection.num_entities)
```


### Milvus Log

_No response_

### Anything else?

_No response_

--- a/internal/proxy/task_insert.go
+++ b/internal/proxy/task_insert.go
@@ -163,6 +163,30 @@ func (it *insertTask) checkPrimaryFieldData() error {
 	return nil
 }
 
+// currently we don't support inserting empty string(not sure the string index can handle it).
+// this method check each row of each varchar field, return error if found any empty string.
+// this is not a good solution, the correct way is:
+//     allow user to insert empty string, store empty string in parquet, ignore empty string in string index
+// TODO: once we support empty string, delete this method
+func (it *insertTask) checkVarcharFieldData() error {
+	fieldsData := it.InsertRequest.GetFieldsData()
+	for _, fd := range fieldsData {
+		if fd.GetType() != schemapb.DataType_VarChar {
+			continue
+		}
+
+		values := fd.GetScalars().GetStringData().GetData()
+		for _, str := range values {
+			if len(str) == 0 {
+				log.Error("value of varchar field cannot be empty string", zap.String("field name", fd.GetFieldName()))
+				return fmt.Errorf("value of varchar field '%s' cannot be empty string", fd.GetFieldName())
+			}
+		}
+	}
+
+	return nil
+}
+
 func (it *insertTask) PreExecute(ctx context.Context) error {
 	sp, ctx := trace.StartSpanFromContextWithOperationName(it.ctx, "Proxy-Insert-PreExecute")
 	defer sp.Finish()
@@ -228,6 +252,13 @@ func (it *insertTask) PreExecute(ctx context.Context) error {
 	err = it.checkPrimaryFieldData()
 	if err != nil {
 		log.Error("check primary field data and hash primary key failed", zap.Int64("msgID", it.Base.MsgID), zap.String("collection name", collectionName), zap.Error(err))
+		return err
+	}
+
+	// check empty string for varchar field. currently we don't support empty string
+	// TODO: once we support empty string, delete this line and the method checkVarcharFieldData()
+	err = it.checkVarcharFieldData()
+	if err != nil {
 		return err
 	}
 

--- a/internal/proxy/task_insert_test.go
+++ b/internal/proxy/task_insert_test.go
@@ -1,6 +1,7 @@
 package proxy
 
 import (
+	"strconv"
 	"testing"
 
 	"github.com/milvus-io/milvus/internal/proto/commonpb"
@@ -345,4 +346,99 @@ func TestInsertTask_CheckAligned(t *testing.T) {
 	case2.FieldsData[8] = newScalarFieldData(varCharFieldSchema, "VarChar", numRows)
 	err = case2.CheckAligned()
 	assert.NoError(t, err)
+}
+
+func TestInsertTask_checkVarcharFieldData(t *testing.T) {
+	ids := make([]int64, 10)
+	for i := 0; i < 10; i++ {
+		ids[i] = int64(i)
+	}
+
+	values := make([]string, 10)
+	for i := 0; i < 10; i++ {
+		values[i] = strconv.Itoa(i)
+	}
+
+	it := insertTask{
+		schema: &schemapb.CollectionSchema{
+			Name:        "TestInsertTask_checkVarcharFieldData",
+			Description: "TestInsertTask_checkVarcharFieldData",
+			AutoID:      false,
+			Fields: []*schemapb.FieldSchema{
+				{
+					AutoID:   true,
+					DataType: schemapb.DataType_Int64,
+				},
+				{
+					AutoID:   false,
+					DataType: schemapb.DataType_VarChar,
+				},
+			},
+		},
+	}
+
+	it.FieldsData = []*schemapb.FieldData{
+		{
+			Type:      schemapb.DataType_Int64,
+			FieldName: "aaa",
+			Field: &schemapb.FieldData_Scalars{
+				Scalars: &schemapb.ScalarField{
+					Data: &schemapb.ScalarField_LongData{
+						LongData: &schemapb.LongArray{
+							Data: ids,
+						},
+					},
+				},
+			},
+		},
+		{
+			Type:      schemapb.DataType_VarChar,
+			FieldName: "bbb",
+			Field: &schemapb.FieldData_Scalars{
+				Scalars: &schemapb.ScalarField{
+					Data: &schemapb.ScalarField_StringData{
+						StringData: &schemapb.StringArray{
+							Data: values,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	err := it.checkVarcharFieldData()
+	assert.Nil(t, err)
+
+	values = append(values, "")
+	it.FieldsData = []*schemapb.FieldData{
+		{
+			Type:      schemapb.DataType_Int64,
+			FieldName: "aaa",
+			Field: &schemapb.FieldData_Scalars{
+				Scalars: &schemapb.ScalarField{
+					Data: &schemapb.ScalarField_LongData{
+						LongData: &schemapb.LongArray{
+							Data: ids,
+						},
+					},
+				},
+			},
+		},
+		{
+			Type:      schemapb.DataType_VarChar,
+			FieldName: "bbb",
+			Field: &schemapb.FieldData_Scalars{
+				Scalars: &schemapb.ScalarField{
+					Data: &schemapb.ScalarField_StringData{
+						StringData: &schemapb.StringArray{
+							Data: values,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	err = it.checkVarcharFieldData()
+	assert.Error(t, err)
 }

